### ISSUE TITLE
Documenting expected directory structure

### ISSUE THREAD

**Issue by JRMeyer** [codalab/codalab-competitions#3019]
:wave: hello CodaLab Maintainers!

Firstly, this is a great project you've open-sourced, providing real value to the ML/DL communities... thank you!

I'm setting up my own competition, first trying to get things working on the public codalab instance at `competitions.codalab.org`. Eventually I want to self-host, but first I want to get a competition working.

My question relates to file structure of the Docker images in which of bundles are run. I've read everything I can find in the documentation, including:

1. [the Medium article](https://towardsdatascience.com/how-to-create-a-competition-on-codalab-85d69580f9c8)
2. [the wiki docs on competition bundles](https://github.com/codalab/codalab-competitions/wiki/User_Building-a-Competition-Bundle)
3. [the wiki docs on ingestion bundles](https://github.com/codalab/codalab-competitions/wiki/User_Building-an-Ingestion-Program-for-a-Competition)
4. [the wiki docs on scoring bundles](https://github.com/codalab/codalab-competitions/wiki/User_Building-a-Scoring-Program-for-a-Competition)

But I'm still confused (and getting submission errors) from the assumed dir structure of the submission evaluation Docker containers.

There appears to be a hard-coded directory structure in the working dir where the bundles are run, including (at least) `input`, `output`, and `program`. I see these terms appear in `metadata` files for scoring programs, along with `$`, so they look like `$input`, `$output`, and `$program`. I thought the `$` was for variable expansion, and these names were arbitrary, but if I change them the program fails. It also looks like `ref` and `res` are subdirs created at runtime.

I added some logging to the `Compute Pi` scoring program (i.e. `score.py`), and I see that the dir structure in the Docker container looks like this

```
codalab/
└── tmp2PvMeL/
    └── run/
        ├── input/
        │   ├── coopetition/
        │   ├── history/
        │   ├── ref/
        │   ├── res/
        │   └── scores/
        ├── output/
        ├── private_output/
        └── program/
```

where `tmp2PvMeL` is some (I assume) randomly generated submission number.

**TLDR - where can I read documentation about the directory structure and functionality of the Docker containers used to evaluate submissions?**

Thanks!

**Comment by Didayolo** [comment]
Hello,

Some information that could help you:
- First of all, as a competition organizer, you don't necessarily need to understand the underlying structure of compute workers. I'm not saying you should not, but it's not mandatory. What are the errors you encounter with your submissions?
- If you [setup your own compute worker](https://github.com/codalab/codalab-competitions/wiki/User_Using-your-own-compute-workers), you'll be able to execute bash inside the containers in order to see it by yourself.
- The code and configuration of the workers can be found in this repository: [https://github.com/codalab/competitions-v1-compute-worker](https://github.com/codalab/competitions-v1-compute-worker).
- Just to clarify in case it's needed. There are several things: (1) the code and dockers of the platform itself, (2) the code and dockers of the compute workers and (3) the ingestion/scoring programs code provided by the challenge organizers.

To give you more details about the directory structure, the metadata files, etc. I'll have to look at it myself first, sorry!
I hope this can already help you a bit. Feel free to ask more questions.

**Comment by JRMeyer** [comment]
thanks for the response @Didayolo !

Unfortunately, there is some knowledge of dir structure required for building a scorer program, as hard coded in the Compute Pi example [here](https://github.com/codalab/competition-examples/blob/04cbbd18ec712674f3cb4a03e84cf1732565506c/basic-competition-bundles/Compute_pi/compute_pi_competition_bundle/program/evaluate.py#L7-L8)

Knowing that the user submission is migrated to `input/res` and reference data is migrated to `input/ref` seems to have solved my issues for now, since I've now coded them in my version of `score.py`

I'll close the issue, now that I know the dir structure which is assumed by the scoring program.

Thanks!

**Comment by Didayolo** [comment]
Good to hear that you've solved your problem!

I'm not sure if it's different there, but the [Iris challenge example](https://github.com/codalab/competition-examples/tree/master/basic-competition-bundles/Iris) is more up to date.
